lib/runtime/src/pipeline/network/egress/addressed_router.rs

```diff
@@ -221,11 +222,18 @@ where
                     }
                 }
             } else if is_complete_final {
+                // end of stream
+                None
+            } else if engine_ctx_.is_stopped() {
+                // Gracefully end the stream if 'stop_generating()' was called. Do NOT check for
+                // 'is_killed()' here because it implies the stream ended abnormally which should be
```

Comment:
are you sure this is the case?

Reply:
For lease "dynamo", I don't get when etcd gets a lease update. Workers will come up with a new lease id, the old expires, and the kv-router re-indexes. Is that correct?

Reply:
(case 1) If the worker is up for the entire time, meaning the lease only gets renewed, the lease id does not change.

(case 2) If the worker process is stopped and then restarted, it will get a new lease id.

I did an experiment on what happen if the network to the worker is disconnected but the process is left running: 
The process found it cannot reach ETCD and then exits. If the process is later restarted, the worker will get a new lease id, same as (case 2).